lldb/source/Target/StackID.cpp

```diff
@@ -76,11 +76,34 @@ bool lldb_private::operator!=(const StackID &lhs, const StackID &rhs) {
 }
 
 // BEGIN SWIFT
+// Given two async contexts, source and destination, chase continuation pointers
+// to check if `destination` can be reached from `source`. The search stops when
+// it hits the end of the chain (parent_ctx == 0) or a safety limit in case of
+// an invalid continuation chain.
+static llvm::Expected<bool> ChaseContinuationsToFind(lldb::addr_t source,
+                                                     lldb::addr_t destination,
+                                                     Process &process) {
+  auto max_num_frames = 512;
+  for (lldb::addr_t parent_ctx = source; parent_ctx && max_num_frames;
+       max_num_frames--) {
+    Status error;
+    lldb::addr_t old_parent_ctx = parent_ctx;
+    // The continuation's context is the first field of an async context.
+    parent_ctx = process.ReadPointerFromMemory(old_parent_ctx, error);
+    if (error.Fail())
+      return llvm::createStringError(llvm::formatv(
+          "Failed to read parent async context of: {0:x}", old_parent_ctx));
+    if (parent_ctx == destination)
+      return true;
+  }
+  return false;
```

Comment:
`false` is correct for the case where a null parent is reached. What about when the 512 limit is reached? Should this return an error? If not, should there be a log? As it is, there's no way of knowing the limit was reached.

Reply:
I can do that, though I was hoping to just move the code as this is an NFC patch

Reply:
Tried it just now, it does look much better indeed!